### DISCUSSION TITLE
page length

### DISCUSSION THREAD

**Discussion by 22syamsudar** [rappasoft/laravel-livewire-tables#251] [Q&A]
How can we show all records by page length? now we just can se 10,25, and 50 records...
how about all records?

**Comment by rappasoft** [comment]
https://github.com/rappasoft/laravel-livewire-tables/wiki/Setting-per-page-options

Requires v1.4

**Comment by 22syamsudar** [comment]
but i tried, it's not working btw..give the example please... i put it on
my componen like... below...


class Detail extends DataTableComponent
{  
    public array $perPageAccepted = [10, 25, 50];
	public bool $perPageAll = true;
}

**Reply by rappasoft** [reply]
Make sure you have the most up to date version of the view files.

**Reply by 22syamsudar** [reply]
oh i see... i just update... thanks... let me improve...

**Reply by 22syamsudar** [reply]
anyway...
its working now...
but it shows just -1 not All...
any suggest?

**Reply by rappasoft** [reply]
The view is still not updated. https://github.com/rappasoft/laravel-livewire-tables/blob/master/resources/views/bootstrap-4/includes/per-page.blade.php#L9

**Reply by rappasoft** [reply]
Delete the resources/vendor/livewire-tables folder if you don't need it.

**Reply by 22syamsudar** [reply]
ok thank you..
well done... 
this is the best vendor of livewire for datatable...
 keep  it  strong...

**Reply by tuluong** [reply]
@22syamsudar btw, is there a way to set the default perPage value? thanks

**Reply by 22syamsudar** [reply]
put this code in you DataTableComponent

    public array $perPageAccepted = [10, 25, 50];

**Reply by rappasoft** [reply]
Set $perPage to an item in your $perPageAccepted if you don't want the first as the default.
